ThingIFSDK/ThingIFSDK/TriggeredServerCodeResult.swift

```diff
@@ -35,39 +35,188 @@ public class TriggeredServerCodeResult: NSObject, NSCoding {
+                return false
+            }
+        } else if self.returnedValue is NSNumber {
+            if self.returnedValue as! NSNumber != aResult.returnedValue as! NSNumber {
+                return false
+            }
+        }
+        if self.error == nil || aResult.error == nil {
+            if self.error != nil || aResult.error != nil {
+                return false
+            }
+        } else if (!self.error!.isEqual(aResult.error!)) {
+            return false
+        }
+        return self.succeeded == aResult.succeeded && self.executedAt == aResult.executedAt
+    }
+    private func isEqualArray(arr1:[AnyObject], arr2:[AnyObject]) -> Bool {
+        if arr1.count != arr2.count {
+            return false
+        }
+        for var i = 0; i < arr1.count; ++i {
+            let e1 = arr1[i]
+            let e2 = arr2[i]
+            if e1 is Dictionary<String, AnyObject> {
+                if !NSDictionary(dictionary: e1 as! [NSObject : AnyObject]).isEqualToDictionary(e2 as! [NSObject : AnyObject]) {
+                    return false
+                }
+            } else if e1 is [AnyObject] {
+                if !isEqualArray(e1 as! [AnyObject], arr2: e2 as! [AnyObject]) {
+                    return false
+                }
+            } else if e1 is String {
+                if e1 as! String != e2 as! String {
+                    return false
+                }
+            } else if e1 is Bool {
+                if e1 as! Bool != e2 as! Bool {
+                    return false
+                }
+            } else if e1 is NSNumber {
+                if e1 as! NSNumber != e2 as! NSNumber {
+                    return false
+                }
+            } else {
+                return false
+            }
+        }
+        return true
     }
-
     
     class func resultWithNSDict(resultDict: NSDictionary) -> TriggeredServerCodeResult?{
         let succeeded = resultDict["succeeded"] as? Bool
-        let returnedValue = resultDict["returnedValue"] as? String
+        let returnedValue = resultDict["returnedValue"] as AnyObject?
         let executedAtStamp = resultDict["executedAt"] as? NSNumber
-        let errorMessage = resultDict["errorMessage"] as? String
+        let error = resultDict["error"] as? Dictionary<String, AnyObject>
+        var serverError: ServerError? = nil
+        if error != nil {
+            serverError = ServerError.errorWithNSDict(error!)
+        }
 
         let executedAt = NSDate(timeIntervalSince1970: (executedAtStamp?.doubleValue)!/1000.0)
         
         var result: TriggeredServerCodeResult?
         if succeeded != nil {
-            result = TriggeredServerCodeResult(succeeded:succeeded!, returnedValue:returnedValue, executedAt:executedAt, errorMessage:errorMessage)
+            result = TriggeredServerCodeResult(succeeded:succeeded!, returnedValue:returnedValue, executedAt:executedAt, error:serverError)
```

Comment:
`succeeded` and `executedAt` are required NonNullable, better to guard them at the beginning of this func.